--- File.pm.orig	Thu Apr 12 15:37:08 2001
+++ File.pm	Thu Apr 12 15:37:36 2001
@@ -133,7 +133,7 @@
     my @files = readdir(DIR);
     foreach my $file (@files) {
         if ($file =~ /^Apache-Session.*\.lock$/) {
-            if ((stat($dir.'/'.$file))[8] - $now >= $time) {
+            if ($now - (stat($dir.'/'.$file))[8] >= $time) {
                 open(FH, "+>$dir/".$file) || next;
                 flock(FH, LOCK_EX) || next;
                 unlink($dir.'/'.$file) || next;
